meta-resin-common/recipes-core/images/resin-image-flasher.bb

```diff
@@ -37,7 +37,11 @@ RESIN_STATE_FS_LABEL = "flash-state"
 RESIN_DATA_FS_LABEL = "flash-data"
 
 # Put the resin logo, uEnv.txt files inside the boot partition
-RESIN_BOOT_PARTITION_FILES_append = " resin-logo.png:/splash/resin-logo.png"
+RESIN_BOOT_PARTITION_FILES_append := " \ 
+    resin-logo.png:/splash/resin-logo.png \
+    ${@ 'resinOS_uEnv.txt:' \
+        if __import__('fnmatch').fnmatch(d.getVar('PREFERRED_PROVIER_' + d.getVar('RESIN_IMAGE_BOOTLOADER', True), True) or  '','u-boot*') else ''} \
```

Comment:
How is this supposed to work? The above line contains a typo, "PREFERRED_PROVIER_"